src/sources/http_client/client.rs

```diff
@@ -58,6 +59,13 @@ pub struct HttpClientConfig {
     #[configurable(metadata(docs::human_name = "Scrape Interval"))]
     pub interval: Duration,
 
+    /// The timeout for each scrape request, in seconds.
+    #[serde(default = "default_target_timeout")]
+    #[serde_as(as = "serde_with::DurationSeconds<u64>")]
+    #[serde(rename = "scrape_target_timeout_secs")]
+    #[configurable(metadata(docs::human_name = "Scrape Target Timeout"))]
+    pub target_timeout: Duration,
```

Comment:
Not sure why this comment I made wasn't saved on the first review but-

Need to run `make generate-component-docs` to generate the user facing docs with this change.

But first:
- I think the usage of "target" may not be necessary. Something like `scrape_timeout` , or just `timeout` might suffice.
- ", in seconds" can be removed, because the auto generated docs will have units due to the macros used.

Reply:
@neuronull renamed this to `scrape_timeout`, but `scrape_interval_secs` uses `_secs` suffix. for consistency, i think i should put that back

Reply:
unless we want to also remove suffix on the interval 😆

Reply:
👍 `_secs` suffix is correct per https://github.com/vectordotdev/vector/blob/master/docs/specs/configuration.md#option-naming